gobblin-runtime/src/main/java/org/apache/gobblin/runtime/fork/Fork.java

```diff
@@ -227,7 +227,10 @@ public void consumeRecordStream(RecordStreamWithMetadata<D, S> stream)
 
           r.ack();
         }
-      }, e -> logger.error("Failed to process record.", e),
+      }, e -> {
+          logger.error("Failed to process record.", e);
+          throw((Exception) e);
```

Comment:
Sorry - I did not realize earlier that e is a throwable, in which case I now understand why you had to wrap it into a RuntimeException. I think that would be better than casting to an Exception.